### PR TITLE
MF-103, redirect back after login

### PR DESCRIPTION
**Pull request by cioan**
Hi @brandones , this PR is related to the PR#21 on the openmrs-esm-primary-navigation. Thanks!

### CODE REVIEW

**Comment by brandones** [comment]
Nice work! Let's also add a couple tests? Maybe just check that, for each of Login and ChooseLocation, if `location.state.referrer = "/foo"` is passed, then the user is sent to `/foo`?

**Comment by brandones** [comment]
Oooh this collided pretty seriously with my PR. Let's resolve this together on Monday.

**Comment by cioan** [comment]
Hi @brandones , i resolved the merge conflicts and I also added test cases for redirecting after login. Thanks!

File: src/login/login.component.test.tsx [hunk 1]
```diff
@@ -94,4 +98,30 @@ describe(`<Login />`, () => {
     await wait();
     expect(wrapper.history.location.pathname).toBe("/home");
   });
+
+  it(`after login with exactly one login location due to timeout it should return back to the current location`, async () => {
```
Comment (brandones) [inline src/login/login.component.test.tsx]:
```suggestion
  it(`should redirect back to referring page on successful login when there is only one location`, async () => {
```

File: src/login/login.component.tsx [hunk 2]
```diff
@@ -197,7 +208,11 @@ export default function Login(props: LoginProps) {
 
 function navigate(props, urlConfig: UrlConfig) {
   if (urlConfig.spa) {
-    props.history.push(urlConfig.url);
+    props.history.push(
+      props.location && props.location.state && props.location.state.referrer
```
Comment (brandones) [inline src/login/login.component.tsx]: This causes the referrer redirect to only work if the configured loginSuccess redirect page is a SPA page. This logic should be totally independent. I would leave this function as it was and instead embed the logic about `location.state.referrer` into your code above.

**Comment by cioan** [comment]
Thanks @brandones . I made the changes requested above. Thanks.

**Pull request merged**